cmd/dp-observation-extractor/main.go

```diff
@@ -33,29 +56,84 @@ func main() {
 		config.FileConsumerTopic,
 		config.FileConsumerGroup,
 		kafka.OffsetNewest)
-	if err != nil {
-		log.Error(err, log.Data{"message": "failed to create kafka consumer"})
-		os.Exit(1)
-	}
+	checkForError(err)
 
-	kafkaProducer := kafka.NewProducer(config.KafkaAddr, config.ObservationProducerTopic, 0)
+	kafkaObservationProducer, err := kafka.NewProducer(config.KafkaAddr, config.ObservationProducerTopic, 0)
+	checkForError(err)
 
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	kafkaErrorProducer, err := kafka.NewProducer(config.KafkaAddr, config.ErrorProducerTopic, 0)
```

Comment:
Not currently - it will be used in a future task for sending errors to the reporter. I intentionally added it now to ensure it was closed on shutdown.